### DISCUSSION TITLE
Database updates hangs "Creating keys until 100...."

### DISCUSSION THREAD

**Discussion by imadevel** [nilsteampassnet/TeamPass#3724] [Q&A] [answered]
Hello,
I wanted to upgrade the system from version 2.1.27.36 to the actual version 3.0.6.
After upgrading Ubuntu, now using Ubuntu 20.04, and installing the missing packages (php-bcmath and php-gmp) I inserted 
define("SECUREPATH", "/includes");
into includes/config/settings.php

Due to the long usage-history in /includes there are two files:
- sk.php.2020_07_27.1595861785 and
- teampass-seckey.txt

The file
-  teampass-seckey.txt contains a long string "def000........." and 
- sk.php.2020_07_27.1595861785 contains
```
<?php
@define('SALT', '090.......'); //Never Change it once it has been used !!!!!
@define('COST', '13'); // Don't change this.
?>
```

Starting the upgrade hangs at "User 9999997 - Creating keys until 100 ..." forever, there is no communication between browser and server (tested with tcpdump).

Even if I rename sk.php.2020_07_27.1595861785 to sk.php the problem stays.
There were no error messages in the apache-log.

Do you have any idea?

Thanks a lot,
Martin

**Comment by WMedDougJaquays** [comment, marked as answer]
See #3700

**Reply by imadevel** [reply]
Thanks, the column encrypted_psk was "NOT NULL" so I did
`
mysql -h localhost -u root -p
connect DATABASENAME;
ALTER TABLE teampass_users MODIFY encrypted_psk text;
quit
`
